integration-tests/src/dust.rs

```diff
@@ -0,0 +1,124 @@
+#![cfg(test)]
+
+use crate::polkadot_test_net::*;
+use frame_support::{assert_ok, sp_runtime::traits::Zero};
+use hydradx_runtime::{Balances, Tokens, Treasury};
+use orml_traits::{currency::MultiCurrency, GetByKey};
+use xcm_emulator::TestExt;
+
+#[test]
+fn balance_should_be_dusted_when_native_balance_is_below_ed() {
+	TestNet::reset();
+
+	Hydra::execute_with(|| {
+		let hdx_ed = <hydradx_runtime::Runtime as pallet_balances::Config>::ExistentialDeposit::get();
+
+		assert!(!hdx_ed.is_zero());
+
+		let transfer_amount = hdx_ed.checked_sub(1).unwrap();
+
+		// set Treasury balance to ED so it's not dusted
+		assert_ok!(Balances::set_balance(
```

Comment:
dont set ED for treasury account.

the treasury account cannot be dusted. You can assert that as well, that it is possible to have < ED in the treasury account and it wont get dusted.

Reply:
Well, no. The treasury account is dusted if the account doesn't exist, and the total issuance is decreased by the amount.

Reply:
The balance pallet (and the treasury pallet) are trying to `deposit_creating` to the treasury account (which doesn't exist yet) with the amount that is below ED, it results in no-op, and the amount id `Drop`ped.

Reply:
i see , it is strange. it feels like it should be allowed.

Reply:
Yes. The problem is that there is not a whitelist. Just a handler. So they are trying to solve dusting by sending dust somewhere.